### PR TITLE
add require statements to top of file

### PR DESCRIPTION
**Pull request by shicholas**
Based on
https://github.com/rails-api/active_model_serializers/issues/1170#issuecomment-141184047

### CODE REVIEW

File: lib/active_model/serializer.rb [hunk 1]
```diff
@@ -1,14 +1,14 @@
 require 'thread_safe'
+require_relative 'serializer/configuration'
+require_relative 'serializer/associations'
```
Comment (bf4) [inline lib/active_model/serializer.rb]: I'd rather use the full require `active_model/serializer/configuration` without further discussion

**Comment by bf4** [comment]
Awesome, thanks!

File: lib/active_model/serializer.rb (hunk 1 above)
Comment (NullVoxPopuli) [inline lib/active_model/serializer.rb]: :+1:

Comment (shicholas) [inline lib/active_model/serializer.rb]: np I'll update this.

**Comment by NullVoxPopuli** [comment]
Is there any sort of test you can add for this @shicholas, so your #1170 doesn't happen in the future?

**Comment by shicholas** [comment]
good question, I'm kind of stuck but I can't think of a test that won't introduce a ton of test dependencies (e.g. a ``test/dummy`` rails app that is present in rails engines). Any thoughts? I'm happy to implement it.

**Comment by shicholas** [comment]
perhaps we can get rid of all ``autoload`` like @bf4 mentioned here: http://www.benjaminfleischer.com/2013/07/18/ruby-requires-confusion/.

**Comment by NullVoxPopuli** [comment]
perhaps. I could get on board with that, even if there is a small performance and memory penalty in certain cases.
@beauby, @joaomdmoura thoughts on that?

**Comment by bf4** [comment]
> perhaps we can get rid of all autoload like @bf4 mentioned here: http://www.benjaminfleischer.com/2013/07/18/ruby-requires-confusion/.

Nice, link!  I should note that Rubies 2.0+ are threadsafe with autoload ref: https://github.com/headius/thread_safe/issues/15#issuecomment-28936120

Here's some links on require vs. require_relative performance in rspec

- http://rubyforge.org/pipermail/rspec-users/2011-November/020760.html
- https://github.com/rspec/rspec-support/pull/45
- https://github.com/rspec/rspec-core/pull/1348
- https://github.com/rspec/rspec-core/pull/502

**Comment by bf4** [comment]
I'm :+1: on getting rid of the autoload, but I'm okay with lazy loading

**Comment by jfelchner** [comment]
I would :heart: getting rid of the autoloads.  Also `require_relative` gives almost [no performance improvement](https://github.com/rspec/rspec-core/commit/af5f60aba5a54bb33be0af196ab0a727319dc15c) to `require` so I think we should just go with `require` as `require_relative` has its own problems.

**Comment by shicholas** [comment]
I pushed an update to use all require statements. Thanks for the links, @bf4, I tried playing around w/ some autoload stuff to no avail (could be user error :)).

**Comment by NullVoxPopuli** [comment]
Well, I'd say if we are going to do away with autoload, and since the issue was with autoload, and that coming up up with a proper test would be really weird / tricky / hacky, I think this in good to go as in. Good work!

**Comment by shicholas** [comment]
I'm having a tough time deciphering the build error messages; it looks like ``simplecov``?

**Comment by NullVoxPopuli** [comment]
Our appveyer build breaks 80% of the time. We're working on it. Lol

**Comment by shicholas** [comment]
sorry, I realized that the test failures were my fault b/c I only removed the ``autoload`` statements in ``lib/active_model/serializer.rb`` and not elsewhere. This is now updated and should now pass.

**Comment by bf4** [comment]
Note of caution, we're not actually using Kernel.autoload but ActiveSupport::Autoload, which *cannot* co-exist with require.  Anyhow once we're using only requires, it won't matter.

Also, requires should *always* be one direction, never circular

**Comment by NullVoxPopuli** [comment]
:+1:

**Pull request merged**